sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -678,4 +663,99 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
       super.afterAll()
     }
   }
+
+  /**
+   * Consumes contents from a single golden file and compares the expected results against the
+   * output of running a query.
+   */
+  def readGoldenFileAndCompareResults(
+    resultFile: String,
```

Comment:
nit: 4 spaces indentation for parameters.

Reply:
Done.